src/main/java/com/vaadin/training/bugrap/ui/MainView.java

```diff
@@ -125,6 +135,16 @@ private void initSearchReportsDropdown() {
 		headerSecondLine.addComponent(searchReportsDropDown);
 	}
 
+	private void initReportView() {
+		reportView.setVisible(false);
+
+		final BugrapApplicationModel applicationModel = getApplicationModel();
+		if (applicationModel.isSingleReportSelected()) {
```

Comment:
Other way would be to remove the reportView from current layout if no report is selected. This would mean setting the lower part of the split screen null.

Reply:
I deleted my previous comment because it totally missed the point ;D

I added the reportView to the design. That's why I thought it would be better just to hide or show it, without changing the structure of the view.